django/contrib/admin/views/main.py

```diff
@@ -35,6 +36,16 @@
     ALL_VAR, ORDER_VAR, ORDER_TYPE_VAR, SEARCH_VAR, IS_POPUP_VAR, TO_FIELD_VAR)
 
 
+class ChangeListForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        super(ChangeListForm, self).__init__(*args, **kwargs)
+        self.fields = {
+            SEARCH_VAR: forms.CharField(required=False, strip=False, initial=''),
```

Comment:
I'm a bit sketpical that we wouldn't want to to strip the search. Were there failing tests? I think `initial=''` isn't needed because that's the default, isn't it?

Reply:
strip is @charettes suggestion because of backward compatiblity.
as I can see the initial is None. At the current code, I've set SEARCH_VAR with init to make sure convert this line properly.
`self.query = request.GET.get(SEARCH_VAR, '')`

Reply:
I'm not quite following the BC concern...

Reply:
@timgraham I could understand the problem about default value.when I check the error that @carltongibson showed. I've changed these and remove initial.

```
self.page_num = change_list_form.cleaned_data.get(PAGE_VAR) or 0
self.query = change_list_form.cleaned_data.get(SEARCH_VAR) or ''
```